CompatibilityReport/Reporter/TextReport.cs

```diff
@@ -496,36 +497,66 @@ private string Statuses(Mod subscribedMod, bool authorRetired)
+                //}
+                //else if (subscribedMod.Statuses.Contains(Enums.Status.MusicCopyrightUnknown))
+                //{
+                //   text += Format("This includes music with unknown copyright status. Safer not to use it for streaming.");
+                //}
             }
 
             if (subscribedMod.Statuses.Contains(Enums.Status.SavesCantLoadWithout))
             {
                 text += Format("NOTE: After using this mod, savegames won't (easily) load without it anymore.");
             }
+                //added to inform about non public source code
+
+            if (subscribedMod.Statuses.Contains(Enums.Status.SourceNotPublic))
+            {
+                text += Format("No source code published. Without source code it is harder to ensure a mod not include malicious code.");
+            }
+
+            // added to inform about mods that have no comment section active, no workshop description and no source code available
+            if (subscribedMod.Statuses.Contains(Enums.Status.NoDescription) && !subscribedMod.Statuses.Contains(Enums.Status.NoCommentSection) && !subscribedMod.Statuses.Contains(Enums.Status.SourceNotPublic))
+            {
+                text += Format("Mods without source codes, without comment section and with an unclear description should better not be subscribed. Without source code it is also harder to ensure a mod not include malicious code.");
+            }
+            else if (subscribedMod.Statuses.Contains(Enums.Status.NoCommentSection) && !subscribedMod.Statuses.Contains(Enums.Status.SourceNotPublic))
+            {
+                text += Format("Mods without source codes and without comment section should better not be subscribed. Without source code it is also harder to ensure a mod not include malicious code.");
+            }
+            else if (subscribedMod.Statuses.Contains(Enums.Status.NoDescription) && !subscribedMod.Statuses.Contains(Enums.Status.SourceNotPublic))
+            {
+                text += Format("Mods without source codes and with an unclear description should better not be subscribed. Without source code it is also harder to ensure a mod not include malicious code.");
+            }
 
             bool abandoned = subscribedMod.Statuses.Contains(Enums.Status.Obsolete) || subscribedMod.Statuses.Contains(Enums.Status.Deprecated) ||
                 subscribedMod.Statuses.Contains(Enums.Status.RemovedFromWorkshop) || subscribedMod.Statuses.Contains(Enums.Status.Abandoned) ||
                 (subscribedMod.Stability == Enums.Stability.IncompatibleAccordingToWorkshop) || authorRetired;
 
-            if (abandoned && string.IsNullOrEmpty(subscribedMod.SourceUrl) && !subscribedMod.Statuses.Contains(Enums.Status.SourceBundled))
+            if (abandoned && string.IsNullOrEmpty(subscribedMod.SourceUrl) && subscribedMod.Statuses.Contains(Enums.Status.SourceBundled))
```

Comment:
missing `!`?
it means: **if** abandoned **and** no source url **and** has status `SourceBundled`

Reply:
btw, that file is text report :)

Reply:
> missing `!`? it means: **if** abandoned **and** no source url **and** has status `SourceBundled`

Yes, as i Not found one Mod that have the `SourceBundled` flag, that really include the Code as ZIP/uncompiled Data.

Reply:
> btw, that file is text report :)

Yeah, i need to Change both files, the HTML as Well as the txt... Or?

Reply:
yeah, both or text report will be different
If you don't need `SourceBundled` flag then maybe don't check for it?

Reply:
I thought about it, but maybe one time in Future one will use. And then i would only need to change the text into something more positive. E.g. i hope with this to get M.S.Referee to deliver His source Code bundeled, as he is worried to release it in other platforms than Steam.